common/djangoapps/embargo/middleware.py

```diff
@@ -181,6 +216,79 @@ def _is_embargoed_by_profile_country(self, user, course_id="", course_is_embargo
         else:
             return None
 
+    def _is_embargoed_by_ip_new(self, ip_addr, course_id=u"", course_is_embargoed=False):
+        """
+        Check whether the user is embargoed based on the IP address.
+
+        Args:
+            ip_addr (str): The IP address the request originated from.
+
+        Keyword Args:
+            course_id (unicode): The course the user is trying to access.
+            course_is_embargoed (boolean): Whether the course the user is accessing has been embargoed.
+
+        Returns:
+            A unicode message if the user is embargoed, otherwise `None`
+
+        """
+        # If blacklisted, immediately fail
+        if ip_addr in IPFilter.current().blacklist_ips:
+            return self.REASONS['ip_blacklist'].format(
+                ip_addr=ip_addr,
+                from_course=self._from_course_msg(course_id, course_is_embargoed)
+            )
+
+        # If we're white-listed, then allow access
+        if ip_addr in IPFilter.current().whitelist_ips:
+            return None
+
+        # Retrieve the country code from the IP address
+        # and check it against the list of embargoed countries
+        ip_country = self._country_code_from_ip(ip_addr)
+        if course_id and CountryAccessRule().is_course_embargoed_in_country(course_id, ip_country):
```

Comment:
If you're using a class method, you don't need to instantiate the object.  You can just do: `CountryAccessRule.is_course_embargoed_in_country`